diff --git NuttX/nuttx/arch/arm/src/stm32f7/chip/stm32f76xx77xx_pinmap.h NuttX/nuttx/arch/arm/src/stm32f7/chip/stm32f76xx77xx_pinmap.h
index b2ded84..d452e30 100644
--- NuttX/nuttx/arch/arm/src/stm32f7/chip/stm32f76xx77xx_pinmap.h
+++ NuttX/nuttx/arch/arm/src/stm32f7/chip/stm32f76xx77xx_pinmap.h
@@ -631,6 +631,7 @@
 #define GPIO_I2C4_SDA_1       (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTD|GPIO_PIN13)
 #define GPIO_I2C4_SDA_2       (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTF|GPIO_PIN15)
 #define GPIO_I2C4_SDA_3       (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTH|GPIO_PIN12)
+#define GPIO_I2C4_SDA_4       (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTB|GPIO_PIN9)
 #define GPIO_I2C4_SMBA_1      (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTD|GPIO_PIN11)
 #define GPIO_I2C4_SMBA_2      (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTF|GPIO_PIN13)
 #define GPIO_I2C4_SMBA_3      (GPIO_ALT|GPIO_AF4 |GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTH|GPIO_PIN10)
diff --git NuttX/nuttx/arch/arm/src/stm32f7/stm32_i2c.c NuttX/nuttx/arch/arm/src/stm32f7/stm32_i2c.c
index 3786e7d..094b028 100644
--- NuttX/nuttx/arch/arm/src/stm32f7/stm32_i2c.c
+++ NuttX/nuttx/arch/arm/src/stm32f7/stm32_i2c.c
@@ -2241,6 +2241,10 @@ static int stm32_i2c_init(FAR struct stm32_i2c_priv_s *priv)
   up_enable_irq(priv->config->er_irq);
 #endif
 
+  /* Force a frequency update */
+
+  priv->frequency = 0;
+
   /* TODO:
    * - Provide means to set peripheral clock source via RCC_CFGR3_I2CxSW
    * - Set to HSI by default, make Kconfig option
